src/smartthings/smartapps/smartthings/amazon-echo-home-skill.src/amazon-echo-home-skill.groovy

```diff
@@ -886,59 +890,85 @@ def deviceHeartbeatCheck() {
-						 case "0063-4944-3033":  // ZWAVE Jasco In-Wall Smart Dimmer (Toggle Style) (12729)
-						 case "0063-5044-3031":  // ZWAVE Jasco Plug-in Smart Dimmer (12718)
-						 case "0063-4944-3034":  // ZWAVE Jasco In-Wall Smart Fan Control (12730)
-							 timeout = 60
-							 break
-					 }
-				 }
-				 break
-		 }
-
-		 // Check DTHs with ambiguous names in type name
-		 if (timeout == 0) {
-			 switch (device.name) {
-				 case "OSRAM LIGHTIFY LED Tunable White 60W":
-					 timeout = 35
-					 break
-			 }
-
-		 }
-	 } catch (Exception e) {
-		 // Catching blanket exception here, only reason is that getData() above is dependent on privileged access and
-		 // we don't want to break device discovery if platform changes are made that breaks above code.
-		 log.error "Heartbeat device lookup failed: $e"
-	 }
-	 return timeout
- }
+	def timeout = 0
+	String timeoutReason = "no heartbeat profile was found for ${device.name} (${device.getTypeName()})"
+	try {
+		String deviceTypeName = device.getTypeName()
+		switch (deviceTypeName) {
+			case "SmartPower Outlet":
+				// works for US and UK SmartPower Outlets
+				timeout = 35
+				timeoutReason = "device type is SmartPower Outlet"
+				break
+			case "Z-Wave Switch":
+			case "Z-Wave Switch Generic":
+			case "Dimmer Switch":
+			case "Z-Wave Dimmer Switch Generic":
+				def msr = "${device?.getZwaveInfo()?.mfr}-${device?.getZwaveInfo()?.prod}-${device?.getZwaveInfo()?.model}"
+				if (msr != null) {
+					switch (msr) {
+						case "001D-1B03-0334":  // ZWAVE Leviton In-Wall Switch (dimmable) (DZMX1-1LZ)
+						case "001D-1C02-0334":  // ZWAVE Leviton In-Wall Switch (non-dimmable) (DZS15-1LZ)
+						case "001D-1D04-0334":  // ZWAVE Leviton Receptacle (DZR15-1LZ)
+						case "001D-1A02-0334":  // ZWAVE Leviton Plug in Appliance Module (Non-Dimmable) (DZPA1-1LW)
+						case "001D-1902-0334":  // ZWAVE Leviton Plug in Lamp Dimmer Module (DZPD3-1LW)
+						case "0063-4952-3031":  // ZWAVE Jasco In-Wall Smart Outlet (12721)
+						case "0063-4952-3033":  // ZWAVE Jasco In-Wall Smart Switch (Toggle Style) (12727)
+						case "0063-4952-3032":  // ZWAVE Jasco In-Wall Smart Switch (Decora) (12722)
+						case "0063-5052-3031":  // ZWAVE Jasco Plug-in Smart Switch (12719)
+						case "0063-4F50-3031":  // ZWAVE Jasco Plug-in Outdoor Smart Switch (12720)
+						case "0063-4944-3031":  // ZWAVE Jasco In-Wall Smart Dimmer (Decora) (12724)
+						case "0063-4944-3032":  // ZWAVE Jasco In-Wall 1000 Watt Smart Dimmer (Decora) (12725)
+						case "0063-4944-3033":  // ZWAVE Jasco In-Wall Smart Dimmer (Toggle Style) (12729)
+						case "0063-5044-3031":  // ZWAVE Jasco Plug-in Smart Dimmer (12718)
+						case "0063-4944-3034":  // ZWAVE Jasco In-Wall Smart Fan Control (12730)
+							timeout = 60
+							timeoutReason = "device type is $deviceTypeName and msr $msr is in the list of heartbeat supported devices"
+							break
+					}
+				}
+				break
+		}
+
+		// def msr = device.device?.getDataValue("MSR")
```

Comment:
Indeed - I only copied it there because I got sidetracked a few times at first, and had to go look it up again.